subsys/bluetooth/host/l2cap.c

```diff
@@ -3247,6 +3264,7 @@ int bt_l2cap_chan_disconnect(struct bt_l2cap_chan *chan)
 	LOG_DBG("chan %p scid 0x%04x dcid 0x%04x", chan, le_chan->rx.cid, le_chan->tx.cid);
 
 	le_chan->ident = get_ident();
+	le_chan->pending_req = BT_L2CAP_DISCONN_REQ;
```

Comment:
Shouldnt't this stuff be done inside the `l2cap_chan_send_req()` function? Otherwise, if something fails (e.g. buffer allocation) or races with this code, then you end up with inconsistent/invalid values inside `le_chan`

Reply:
I agree that this would be cleaner, but then we would need to rework functionality like in `bt_l2cap_ecred_chan_reconfigure`:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/l2cap.c#L3095-L3105
(Sets ident for multiple channels, but `l2cap_chan_send_req` is used only for one channel, meaning it would need to know how many channels to set ident/requested opcode for). A similar instance is in `l2cap_ecred_conn_req` (and `bt_l2cap_ecred_chan_reconfigure_explicit`)